src/com/xilinx/rapidwright/design/DesignTools.java

```diff
@@ -1076,12 +1077,14 @@ public static Set<PIP> getTrimmablePIPsFromPins(Net net, Collection<SitePinInst>
 	            rPips.add(pip);
 	        }
 
-	        Integer count = fanout.get(startNode);
-	        if(count == null){
-	            fanout.put(startNode, 1);
-	        }else{
-	            fanout.put(startNode, count+1);
-	        }
+	        SitePin sp = startNode.getSitePin();
+	        SiteInst si = (sp != null && sp.isInput()) ? design.getSiteInstFromSite(sp.getSite()) : null;
+	        SitePinInst spi = (si != null) ? si.getSitePinInst(sp.getPinName()) : null;
+
+		// If a site pin was found and it belongs to this net, add an extra fanout to
+		// reflect that it was both used for downstream connection as well as this site pin
+	        int fanoutCount = (spi != null && spi.getNet().equals(net)) ? 2 : 1;
```

Comment:
```suggestion
	        int fanoutCount = 1;
	        SitePin sp = startNode.getSitePin();
	        if(sp != null && sp.isInput()) {
	            SiteInst si = design.getSiteInstFromSite(sp.getSite());
	            SitePinInst spi = (si != null) ? si.getSitePinInst(sp.getPinName()) : null;
    
		    // If a site pin was found and it belongs to this net, add an extra fanout to
		    // reflect that it was both used for downstream connection as well as this site pin
	            if(spi != null && spi.getNet().equals(net)) {
	                fanoutCount = 2;
	            }
	        }
```
Sorry for the bad description.  The idea it to optimize for the common case where a PIP does not connect to a SitePin.